--- swift/test/Interop/Cxx/stdlib/use-std-function.swift
+++ swift/test/Interop/Cxx/stdlib/use-std-function.swift
@@ -2,6 +2,16 @@
 // RUN: %target-run-simple-swift(-I %S/Inputs -cxx-interoperability-mode=swift-6)
 // RUN: %target-run-simple-swift(-I %S/Inputs -cxx-interoperability-mode=upcoming-swift)
 
+// libstdc++11 declares a templated constructor of std::function with an rvalue-reference parameter,
+// which aren't yet supported in Swift. Therefore initializing a std::function from Swift closures
+// will not work on the platforms that are shipped with this version of libstdc++ (rdar://125816354).
+// XFAIL: LinuxDistribution=ubuntu-22.04
+// XFAIL: LinuxDistribution=rhel-9.3
+// XFAIL: LinuxDistribution=rhel-9.4
+// XFAIL: LinuxDistribution=fedora-39
+// XFAIL: LinuxDistribution=fedora-40
+// XFAIL: LinuxDistribution=fedora-41
+
 // REQUIRES: executable_test
 
 import StdlibUnittest
